mods/seasonal/moves.js

```diff
@@ -1848,42 +1848,134 @@ let BattleMovedex = {
-		},
-		onHit: function (target, pokemon) {
-			if (!pokemon.transformInto(target, pokemon)) {
-				return false;
-			}
-		},
-		secondary: {
-			chance: 100,
-			volatileStatus: 'flinch',
+		volatileStatus: 'purplepills',
+		effect: {
+			noCopy: true,
+			onStart: function (pokemon) {
+				this.add('-start', pokemon, 'Purple Pills', '[silent]');
+				this.add('-message', `${pokemon.name} downed some pills!`);
+				const allTypes = ['Normal', 'Fire', 'Fighting', 'Water', 'Flying', 'Grass', 'Poison', 'Electric', 'Ground', 'Psychic', 'Rock', 'Ice', 'Bug', 'Dragon', 'Ghost', 'Dark', 'Steel', 'Fairy'];
+				let type1 = allTypes[this.random(18)];
+				let type2 = allTypes[this.random(18)];
+				if (type1 === type2) {
+					pokemon.types = [type1];
+					this.add('-start', pokemon, 'typechange', `${type1}`);
+				} else {
+					pokemon.types = [type1, type2];
+					this.add('-start', pokemon, 'typechange', `${type1}/${type2}`);
+				}
+				// @ts-ignoretrack percentages to keep purple pills from resetting pp
+				pokemon.ppPercentages = pokemon.moveSlots.slice().map(m => {
+					return m.pp / m.maxpp;
+				});
+				// 3 moves with different requirements so no for statement
+				let newMovep = [];
+				let randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Special') continue;
+					if (move.type !== type1) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
+				randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Special') continue;
+					if (move.type !== type2) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
+				randMoves = [];
+				for (let i in exports.BattleMovedex) {
+					let move = exports.BattleMovedex[i];
+					if (i !== move.id) continue;
+					if (move.isZ || move.isNonstandard || !move.isViable) continue;
+					if (move.category !== 'Status') continue;
+					if (move.type !== type1 && move.type !== type2) continue;
+					randMoves.push(move);
+				}
+				newMovep.push(toId(randMoves[this.random(randMoves.length)]));
+				newMovep.push('purplepills');
+				// Update movepool
+				pokemon.moveSlots = [];
+				for (let i = 0; i < newMovep.length; i++) {
+					let moveid = newMovep[i];
+					let move = this.getMove(moveid);
+					if (!move.id) continue;
+					pokemon.moveSlots.push({
+						move: move.name,
+						id: move.id,
+						// @ts-ignore hacky way to reduce purple pill's PP
+						pp: Math.floor(((move.noPPBoosts || move.isZ) ? move.pp : move.pp * 8 / 5) * (pokemon.ppPercentages && move.id === 'purplepills' ? pokemon.ppPercentages[i] : 1)),
```

Comment:
`* (pokemon.ppPercentages ? pokemon.ppPercentages[i] : 1)` will work, your current implementation allows for the recovery of PP with these new moves (ex: these new moves have new PP), which isn't allowed.